src/jukebox/components/playermpd/__init__.py

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*-
+"""
+Package for interfacing with the MPD Music Player Daemon
+"""
+import mpd
+import threading
+import logging
+import time
+import jukebox.cfghandler
+import jukebox.plugs as plugs
+from jukebox.NvManager import nv_manager
+import jukebox.pubsub as pubsub
+
+logger = logging.getLogger('jb.PlayerMPD')
+cfg = jukebox.cfghandler.get_handler('jukebox')
+
+
+class PlayerMPD:
+    """Interface to MPD Music Player Daemon"""
+
+    def __init__(self):
+        self.nvm = nv_manager()
+        self.pubsubserver = pubsub.get_publisher()
+        self.mpd_host = cfg.getn('playermpd', 'host')
+        self.music_player_status = self.nvm.load(cfg.getn('playermpd', 'status_file'))
+
+        self.mpd_client = mpd.MPDClient()
+        self.mpd_client.timeout = 0.5               # network timeout in seconds (floats allowed), default: None
+        self.mpd_client.idletimeout = 0.5           # timeout for fetching the result of the idle command
+        self.connect()
+        logger.info(f"Connected to MPD Version: {self.mpd_client.mpd_version}")
+
+        if not self.music_player_status:
+            self.music_player_status['player_status'] = {}
+            self.music_player_status['audio_folder_status'] = {}
+            self.music_player_status.save_to_json()
+            self.current_folder_status = {}
+        else:
+            last_played_folder = self.music_player_status['player_status'].get('last_played_folder')
+            if last_played_folder is not None:
+                self.current_folder_status = self.music_player_status['audio_folder_status'][last_played_folder]
+                self.mpd_client.clear()
+                self.mpd_client.add(last_played_folder)
+                logger.info(f"Last Played Folder: {last_played_folder}")
+
+        self.old_song = None
+        self.mpd_status = {}
+        self.mpd_status_poll_interval = 0.25
+        self.mpd_mutex = threading.Lock()
+        self.status_is_closing = False
+        self.status_thread = threading.Timer(self.mpd_status_poll_interval, self._mpd_status_poll).start()
+
+    def exit(self):
+        logger.debug("Exit routine of playermpd started")
+        self.status_is_closing = True
+        # Need to make sure we are not in the process of a status poll
+        # This currently causes 250 ms delay in shutdown (which otherwise takes 6 ms)
+        # --> Change it (self.status_thread.cancel() ?)
+        time.sleep(self.mpd_status_poll_interval)
+        # self.status_thread.cancel()
+        self.mpd_client.disconnect()
+        self.nvm.save_all()
+
+    def connect(self):
+        self.mpd_client.connect(self.mpd_host, 6600)
```

Comment:
I tried to run this in my Docker env and I received this error. Could `self.mpd_host` be empty for some reason?

```
jukebox    | 22.07.2021 15:47:29 -  514:plugs.py        - jb.plugin            - MainThread      - ERROR    - Failed to load package: components.playermpd
jukebox    | 22.07.2021 15:47:29 -  515:plugs.py        - jb.plugin            - MainThread      - ERROR    - Reason: AttributeError: 'NoneType' object has no attribute 'startswith'
jukebox    | Traceback (most recent call last):
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/run_jukebox.py", line 56, in <module>
jukebox    |     main()
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/run_jukebox.py", line 52, in main
jukebox    |     myjukebox.run()
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/jukebox/daemon.py", line 99, in run
jukebox    |     plugin.load_all_named(plugins_named, prefix='components')
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/jukebox/plugs.py", line 535, in load_all_named
jukebox    |     raise e
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/jukebox/plugs.py", line 530, in load_all_named
jukebox    |     load(package, load_as, prefix)
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/jukebox/plugs.py", line 516, in load
jukebox    |     raise e
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/jukebox/plugs.py", line 508, in load
jukebox    |     _PLUGINS[load_as].module = importlib.import_module(f'{package}', 'pkg')
jukebox    |   File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
jukebox    |     return _bootstrap._gcd_import(name[level:], package, level)
jukebox    |   File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
jukebox    |   File "<frozen importlib._bootstrap>", line 983, in _find_and_load
jukebox    |   File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
jukebox    |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
jukebox    |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
jukebox    |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/components/playermpd/__init__.py", line 298, in <module>
jukebox    |     player_ctrl = PlayerMPD()
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/components/playermpd/__init__.py", line 30, in __init__
jukebox    |     self.connect()
jukebox    |   File "/home/pi/RPi-Jukebox-RFID/src/jukebox/components/playermpd/__init__.py", line 65, in connect
jukebox    |     self.mpd_client.connect(self.mpd_host, 6600)
jukebox    |   File "/usr/local/lib/python3.7/dist-packages/mpd/base.py", line 782, in connect
jukebox    |     if host.startswith("/"):
jukebox    | AttributeError: 'NoneType' object has no attribute 'startswith'
```

Reply:
Nevermind, I believe I had not updated the `jukebox.yaml`